Sources/SwiftParser/TopLevel.swift

```diff
@@ -189,6 +189,35 @@ extension Parser {
     )
   }
 
+  private mutating func parseStatementItem() -> RawCodeBlockItemSyntax.Item {
+    let stmt = self.parseStatement()
+
+    // Special case: An 'if' or 'switch' statement followed by an 'as' must
+    // be an if/switch expression in a coercion.
+    // We could also achieve this by more eagerly attempting to parse an 'if'
+    // or 'switch' as an expression when in statement position, but that
+    // could result in less useful recovery behavior.
```

Comment:
Wouldn’t this also apply to `is`. Something stupid like

```swift
if true { 1 } else { 2 } is Int
```

Or is `as` the only thing where we allow a postfix expression after an `if` expression?

Reply:
`as` is currently the only thing we're going to allow as a postfix expression (we don't allow `as?` or `as!`, although we'll parse them here, but we'll diagnose them with a semantic error). We could allow `is` here too, but it will be diagnosed with a semantic error instead of a parse error.

Reply:
OK, that makes sense to me. Just wanted to make sure we’re not missing anything.